src/averageAges.js

```diff
@@ -15,11 +15,14 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const getMen = people.filter((person) => person.sex === 'm');
+  const currentMan = century
+    ? getMen.filter((person) => Math.ceil(person.died / 100) === century)
+    : getMen;
+
+  return currentMan
+    .map((person) => person.died - person.born)
+    .reduce((a, b) => a + b) / currentMan.length;
```

Comment:
introduce a helper function that will calculate the average age from a passed array of people